cilium/cmd/bpf_metrics_list.go

```diff
@@ -37,31 +38,85 @@ const (
 	bytesTitle     = "BYTES"
 )
 
+type metricsRow struct {
+	reasonCode int
+	reasonDesc string
+	direction  string
+	packets    int
+	bytes      int
+}
+
+type jsonMetricValues struct {
+	Packets uint64 `json:"packets"`
+	Bytes   uint64 `json:"bytes"`
+}
+
+type jsonMetric struct {
+	Description string                      `json:"description"`
+	Values      map[string]jsonMetricValues `json:"values"`
+}
+
+type jsonMetrics map[int]jsonMetric
+
 var bpfMetricsListCmd = &cobra.Command{
 	Use:   "list",
 	Short: "List BPF datapath traffic metrics",
 	Run: func(cmd *cobra.Command, args []string) {
 		common.RequireRootPrivilege("cilium bpf metrics list")
+		listMetrics(metricsmap.Metrics)
+	},
+}
+
+func listMetrics(m metricsmap.MetricsMap) {
+	bpfMetricsList := make(map[string]string)
+	callback := func(key bpf.MapKey, value bpf.MapValue) {
+		bpfMetricsList[key.String()] = value.String()
```

Comment:
This is a minor thing, but AFAICT here we transform the numbers into strings only to transform them back into numbers for the JSON case. Would it make sense to have a different callback function for the JSON case so that we don't we do the double transformation? I think the resulting code might end up being simpler.

Reply:
Here we are still dealing with entries which have keys like `reason:0 dir:1` and values like `count:63752 bytes:20913476` (see [here](https://github.com/cilium/cilium/blob/master/pkg/maps/metricsmap/metricsmap.go#L104-L122) for the definition of `func (vs Values) String() string` and `func (k Key) String() string`) which need to first get split into the individual values so I don't think there's much we can optimize yet

Reply:
Thanks for the explanation!

It wasn't so much about improving performance but rather to potentially simplify things. Seems this is not as easy as I originally thought.